src/lib/sw.js

```diff
@@ -19,3 +23,11 @@ workbox.routing.registerRoute(
   new RegExp("/([\\w-]+/)*$"),
   new workbox.strategies.StaleWhileRevalidate(),
 );
+
+workbox.routing.setCatchHandler(({event}) => {
```

Comment:
Do we hit this only when the user is offline or does it trigger if any of the routes throws an error? Maybe it's fine, but I was curious if we should double check `navigator.onLine` (even though it's terrible...) and if the user is online but a route just exploded, maybe we give them an 'oops! something went wrong' page? Maybe that'll never happen in practice tho ¯\_(ツ)_/¯

Reply:
This was the advice in the official docs: https://developers.google.com/web/tools/workbox/guides/advanced-recipes#provide_a_fallback_response_to_a_route

Remember that 404's aren't errors. So I think this is safe. Honestly I am probably just going to ask @jeffposnick to look at the site before we launch it :) 👋

Reply:
Right, `catchHandler` will fire whenever `workbox-routing` encounters an exception when trying to generate a response. This would most likely happen due to being offline without an appropriate cache entry being available to fall back to.

I'm super-happy to review the full service worker, and in particular, I can help you move to Workbox v5.

Reply:
> Right, catchHandler will fire whenever workbox-routing encounters an exception when trying to generate a response. This would most likely happen due to being offline without an appropriate cache entry being available to fall back to.

👍

> I'm super-happy to review the full service worker, and in particular, I can help you move to Workbox v5.

Thank you Jeff! I think we're getting pretty close so we'll give you a shout when we hit a stopping point.